packages/babel-plugin-transform-classes/src/index.js

```diff
@@ -21,6 +21,8 @@ export default declare((api, options) => {
 
   const { loose } = options;
 
+  const setClassMethods = options.loose ?? api.assumption("setClassMethods");
```

Comment:
Actually, can we flip this? Have `api.assumption` return `undefined` unless is explicitly specified, then default to `loose`:

```js
api.assumption("setClassMethods") ?? options.loose
```

I think the assumption should be the highest priority, and we fall back to the old loose flag if the assumption isn't set.

Reply:
The rationale for giving precedence to `loose` is that, even if it's "older", it's closer to the plugin:
```json
{
  "assumptions": {
    "noDocumentAll": false
  },
  plugins: [
    "@babel/proposal-nullish-coalescing-operator",
    ["@babel/proposal-optional-chaining", { "loose": true }]
  ]
}
```

I think that the most intuitive thing would be to respect `loose` for `@babel/proposal-optional-chaining` (and thus output `== null`), because it's explicitly passed to that specific plugin.

Reply:
We can deprecate loose option in Babel 8 and rename loose in optional chaining to noDocumentAll. Otherwise it is unclear for users that loose is actually meant to override the noDocumentAll assumption.

Reply:
So the question is if the top level thing (assumptions) should override the one closer to a plugin.. well either way we should notify the user somehow right? We don't want to suggest that both should be on right?

Reply:
IMO, it should go:

```suggestion
  const setClassMethods =
    options.setClassMethods ??
    api.assumption("setClassMethods") ??
    options.loose;
```

Reply:
Half of the goal of the assumptions RFC is to centralise this kind of option rather than keeping it per-plugin, since (especially with class plugins) they have big cross-plugin implications.

Reply:
I agree that `loose` is "closer" to the plugin, but the assumptions list is highly specific to transformer output. So I see it as being the higher precedence item.

Plus, I can see toggling certain assumptions on/off _and_ using `loose` at the same time. This would allow devs to override some of the parts of `loose`, eg enabling `classCallCheck` while otherwise outputting loose code.

Reply:
For backward compatibility to Babel < 7.13, we should check if `api.assumption` is available.